masonry/src/widget/widget.rs

```diff
@@ -220,9 +220,11 @@ pub trait Widget: AsAny {
         None
     }
 
-    // TODO - Document
-    // TODO - Add &UpdateCtx argument
-    fn get_cursor(&self) -> CursorIcon {
+    /// Return the cursor icon for this widget.
+    ///
+    /// **pos** - the position in global coordinates (e.g. `(0,0)` is the top-left corner of the
+    /// window).
+    fn get_cursor(&self, ctx: &QueryCtx, pos: Point) -> CursorIcon {
```

Comment:
The combination of the `cursor_icon_changed` mechanism and this method also taking a position is quite weird.

If you know the cursor has changed, you know that it has entered a specific region.

I can't think of a cleaner solution at the moment, though

Reply:
`cursor_icon_changed` is more for situations where the cursor doesn't move, but the icon does.

I guess I should document that.